### PR TITLE
feat(main.rs): append/use plugin name rather than package filename

### PR DESCRIPTION
**Pull request by vdice**
This allows for an arbitrary package/executable name but ensures the plugin is invoked by its assigned name via spin.

This won't change the behavior when the plugin name and package filename are the same, which might usually be the case, but supports use cases where the filename may differ per the plugin developer's use case.  For example a `foo` plugin whose package filename is `foo-plugin`:

```toml
name = "foo"
version = "0.1"
spin_compatibility = ">=0.7"
license = "Apache-2.0"
package = "./target/release/foo-plugin"
```

Regardless of the package filename, the plugin will be invoked via its given name, here `"foo"`, eg `spin foo bar`.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -214,30 +214,29 @@ impl PluginifyCommand {
...
 
-        let filename = package.file_name().ok_or_else(|| anyhow!("Can't get filename of {}", package.display()))?;
         let tar_path = PathBuf::from(format!("{}-{}-{}-{}.tar.gz", ps.name, ps.version, os, arch)).absolutize()?.to_path_buf();
         if self.verbose {
             eprintln!("About to create tar archive at {}", tar_path.display());
         }
 
         if self.verbose {
-            eprintln!("Appending {} with name {:?}", package.display(), filename);
+            eprintln!("Appending {} with name {:?}", package.display(), ps.name);
```
Comment (itowlson) [inline src/main.rs]: This might need to be Windowsified I think?  (This wasn't an issue with `package.file_name` because a prior step Windowsified that already.)

File: src/main.rs [hunk 2]
```diff
@@ -214,30 +214,29 @@ impl PluginifyCommand {
...
         }
 
         let mut writer = std::fs::File::create(&tar_path)?;
         {
             let mut enc = GzEncoder::new(&writer, Compression::default());
             {
                 let mut tar_builder = tar::Builder::new(&mut enc);
-                tar_builder.append_path_with_name(&package, filename)?;
+                tar_builder.append_path_with_name(&package, &ps.name)?;
```
Comment (itowlson) [inline src/main.rs]: Again I think we need to consider the Windows case?

**Review by itowlson — CHANGES_REQUESTED** [review]
The same thing bit me on the butt all the damn time... _grin_

**Comment by vdice** [comment]
Thanks @itowlson; updated.  Also open to var name alternatives if `os_specific_name` is too wordy or otherwise insufficient.

**Comment by itowlson** [comment]
Sorry @vdice, I wasn't clear - the worry is not about using an OS string - it is that on Windows, the name needs to have a `.exe` suffix.  We mangle the _source_ executable path at https://github.com/itowlson/spin-pluginify/blob/466d7e889a4828421ef12352ce455465933d39f0/src/main.rs#L257 - but with your change we no longer mangle the _tarred_ executable name.  Hope that makes more sense now.

**Comment by vdice** [comment]
I see now (I hope), thank you.  Added a bit of code though eager for pointers on simplifying or a better way to do the job.

**Review by itowlson — APPROVED** [review]
LGTM - thanks!
